enhancements/20240805-encrypted-block-volume.md

```diff
@@ -0,0 +1,335 @@
+# Support Block Volume Encryption/Decryption
+
+This enhancement focuses on the encryption and decryption of block volumes, particularly involving boot volumes and data volumes. 
+
+
+## Summary
+
+This enhancement introduces encryption and decryption capabilities for block volumes, specifically targeting boot volumes (virtual machine images) and data volumes. This functionality will enhance security by ensuring that data stored in these volumes is encrypted and can be decrypted when attached. 
+
+## Related Issues
+
+https://github.com/harvester/harvester/issues/3129
+
+## Motivation
+
+
+### Goals
+
+- Implement encryption for boot volumes and data volumes (support hot plug). 
+- Provide a clear and detailed setup process for users to enable encryption/decryption. 
+
+### Non-goals
+
+- This enhancement will not cover GUI-related changes. That particularly means we won't provide more convenient way to enable encryption/decryption with GUI in this enhancement. This will be addressed in next enhancement.
+
+
+## Proposal
+
+Since Longhorn uses clone mechanism to encrypt/decrypt Boot volume, so Harvester will follow that. This means that there would be two virtual machine image after encryption. One is encrypted, and the other is source virtual machine image. 
+
+![](./20240805-encrypted-block-volume/lh-flow.png)
+
+Data volumes are different from Boot volume. Because Longhorn supports block encryption/decryption, so Harvester could encrypt/decrypt data volume directly. 
+
+Regardless of volumes used, Longhorn will decrypt volume when it's attached to virtual machine. So, we'll be able to read the boot volumes and data volumes in the virtual machine. 
```

Comment:
Can user further create new images/volumes from the running VM ? If user can freely do, it will be a possible security issue.

Reply:
For now, yes. Continue with previous comment, we should encourage users to use RBAC to avoid restricted users from exporting images/volumes from running VMs, encryption or decryption.